src/lib.rs

```diff
@@ -413,26 +472,81 @@ impl UnixStream {
     /// If the provided value is `None`, then `write` calls will block
     /// indefinitely. It is an error to pass the zero `Duration` to this
     /// method.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// use unix_socket::UnixStream;
+    /// use std::time::Duration;
+    ///
+    /// let stream = UnixStream::connect("/path/to/my/socket").unwrap();
+    /// stream.set_write_timeout(Some(Duration::from_millis(1500))).unwrap();
+    /// ```
     pub fn set_write_timeout(&self, timeout: Option<Duration>) -> io::Result<()> {
         self.inner.set_timeout(timeout, libc::SO_SNDTIMEO)
     }
 
     /// Returns the read timeout of this socket.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// use unix_socket::UnixStream;
+    ///
+    /// let stream = UnixStream::connect("/path/to/my/socket").unwrap();
+    /// println!("{}", match stream.read_timeout() {
```

Comment:
The examples seem to be inconsistent as to whether they panic or match on the result.